android/src/main/java/pmstation/MainActivity.java

```diff
@@ -372,11 +372,16 @@ private void showChart() {
     }
 
     private void showPermissionDialog() {
+        Log.i(TAG, "Going to check Bluetooth permission");
         if (!BluetoothLeService.checkPermission(this)) {
-            ActivityCompat.requestPermissions(
-                    this,
-                    new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION},
-                    2);
+            Log.i(TAG, "Going to request Bluetooth permission");
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
```

Comment:
Is it ok, or there is better solution? @sanchin

Reply:
I think it should be BLUETOOTH_SCAN. CONNECT is for already paired devices:
https://developer.android.com/develop/connectivity/bluetooth/bt-permissions